packages/story-editor/src/components/canvas/rightClickMenu.js

```diff
@@ -42,32 +42,23 @@ const RightClickMenuContainer = styled.div`
   z-index: 9999;
 `;
 
-const LabelWrapper = styled.span`
-  width: calc(100% + 14px);
-  display: inline-block;
-  position: relative;
-  margin-left: -14px;
-  svg {
-    width: 32px;
-    position: absolute;
-    top: -25%;
-  }
-  span {
-    padding-left: 32px;
-    font-size: 12px;
-    display: inline-block;
-    white-space: nowrap;
-    overflow: hidden;
-    text-overflow: ellipsis;
-  }
-`;
-
-const SuffixIconWrapper = styled.div`
-  svg {
-    width: 32px;
-    color: ${({ theme }) => theme.colors.fg.secondary};
-    margin: -12px 0 !important;
-  }
+const MenuButton = styled(ContextMenuComponents.MenuButton)`
+  ${({ supportsIcon }) =>
+    supportsIcon &&
+    `
+    svg {
+      width: 32px;
+      position: absolute;
+      margin-left: -12px;
+    }
+    span {
+      padding-left: 18px;
+      font-size: 12px;
+      white-space: nowrap;
+      overflow: hidden;
+      text-overflow: ellipsis;
+    }
+  `}
```

Comment:
💭 Maybe this should be `MenuItemPrefix` in case we want to use this again later?

Reply:
I guess it could be `MenuItemWithPrefix` since it would influence all the buttons of that menu. Or maybe it could add prefix which behaves as a placeholder as well, have to figure out the styling though. I'll check that as the last thing, will try to sort out the hovering part first since that might need a few changes.

Reply:
It's a lot of new stuff for the context menu but I think it's all good stuff. I tend to make a PR to the design system first and then implement it later in the editor

Faster testing and feedback in storybook too

Reply:
@samwhale I've added new Story now for the menu with submenu as well, PTAL (WithSubMenu).

There are probably still things to improve, e.g. currently `firstChild` is used in quite a few places (to get the actual menu, not just the container where the ref is), not sure if there's a better alternative.